23&22/json.py

```diff
@@ -0,0 +1,36 @@
+import sqlite3
+
+print("Creating JSON output on food.js...")
+
+conn = sqlite3.connect('foodfacts.sqlite')
+cur = conn.cursor()
+
+cur.execute('''SELECT data.foodId, foods.name, data.energy, data.fat, data.carbohydrates, data.fiber, data.proteins, data.salt, foods.url 
+    FROM foods JOIN data ON foods.id = data.foodId
+    ''')
+
+def check(data):
+    if data == "?" :
+        return -9999
+    else:
+        return data
+
+fhand = open('food.js','w')
+
+count = 0
+
+#writes in food.js
+fhand.write('spiderJson = {"foods":[\n')
+
+for row in cur :
+    if count > 0 :
+        fhand.write(',\n')
+    fhand.write('{' + '"foodId":' + str(row[0]) + ',"name":"' + str(row[1]) + '",')
+    fhand.write(' "energy":"' + str(check(row[2])) + '", "fat":"' + str(check(row[3])) + '", "carbohydrates":"' + str(check(row[4])) + '",')
```

Comment:
with the `check` function as is or make the change as commented, you should remove the `str( ... )` function call, because number and string is treated differently in JSON.  for example, "value": 123 vs "value": "123"

Reply:
oh I tried that it says can not concate a string with a int value